application/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FeatureToGeoJsonJacksonMapper.java

```diff
@@ -0,0 +1,118 @@
+import java.util.List;
+
+public class FeatureToGeoJsonJacksonMapper {
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    private static final String TYPE = "type";
+    private static final String FEATURE_COLLECTION = "FeatureCollection";
+    private static final String FEATURES = "features";
+    private static final String FEATURE = "Feature";
+    private static final String GEOMETRY = "geometry";
+    private static final String PROPERTIES = "properties";
+
+    public ObjectNode convert(FeatureCollection featureCollection) throws IOException {
+        return buildFeatureCollection(featureCollection);
+    }
+
+    public ObjectNode convert(SimpleFeature feature) throws IOException {
+        return buildFeature(feature);
+    }
+
+    @SuppressWarnings("unchecked")
+    private ObjectNode buildFeatureCollection(FeatureCollection featureCollection) throws IOException {
+        List<ObjectNode> features = new LinkedList<>();
+        FeatureIterator simpleFeatureIterator = featureCollection.features();
+        while (simpleFeatureIterator.hasNext()) {
+            Feature simpleFeature = simpleFeatureIterator.next();
+            features.add(buildFeature((SimpleFeature) simpleFeature));
+        }
+
+        ObjectNode obj = mapper.createObjectNode();
+        obj.put(TYPE, FEATURE_COLLECTION);
+        obj.putArray(FEATURES).addAll(features);
+        return obj;
+    }
+
+    @SuppressWarnings("unchecked")
+    private ObjectNode buildFeature(SimpleFeature simpleFeature) throws IOException {
+        ObjectNode node = mapper.createObjectNode();
+        node.put(TYPE, FEATURE);
+        node.set(GEOMETRY, buildGeometry((Geometry) simpleFeature.getDefaultGeometry()));
+        node.set(PROPERTIES, buildProperties(simpleFeature));
+        return node;
+
+    }
+
+    @SuppressWarnings("unchecked")
+    private ObjectNode buildProperties(SimpleFeature simpleFeature) {
+        ObjectNode obj = mapper.createObjectNode();
+        Collection<Property> properties = simpleFeature.getProperties();
+        for (Property property : properties) {
+            final Object value = property.getValue();
+
+            if (!property.getName().getLocalPart().equals(GEOMETRY)){
+                if (ArrayList.class.equals(property.getType().getBinding())){
+                    ArrayNode arrayNode = mapper.createArrayNode();
+                    for (Object o : (ArrayList)value){
+                        arrayNode.add(o.toString());
+                    }
+                    obj.putArray(property.getName().toString()).addAll(arrayNode);
+                }else if(List.class.equals(property.getType().getBinding())) {
+                    ArrayNode arrayNode = mapper.createArrayNode();
+                    for (Object o : (ArrayList) value) {
+                        arrayNode.add(o.toString());
+                    }
+                    obj.putArray(property.getName().toString()).addAll(arrayNode);
+                }
```

Comment:
The code in the if- and else if-clause are exactly the same

Reply:
Yes, this file is copied from commons-geometry, there might be dragons here, as with any code from the commons dependencies